vgrep.go

```diff
@@ -216,6 +231,51 @@ func getEditorLineFlag() string {
 	return editor
 }
 
+// Create the lock file to guard against concurrent processes
+func makeLockFile() error {
+	var err error
+	lockdir := filepath.Join(os.Getenv("HOME"), ".local/share/vgrep")
+	exists := true
+
+	if _, err := os.Stat(lockdir); err != nil {
+		if os.IsNotExist(err) {
+			exists = false
+		} else {
+			return err
+		}
+	}
+
+	if !exists {
+		if err := os.MkdirAll(lockdir, 0700); err != nil {
+			return err
+		}
+	}
+	lock, err = lockfile.New(filepath.Join(lockdir, "cache-lock"))
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// Try to acquire the lock file for the cache
+func acquireLock() error {
+	err := lock.TryLock()
+
+	// If the lock is busy, wait for it
+	for err == lockfile.ErrBusy {
+		time.Sleep(10 * time.Millisecond)
+		err = lock.TryLock()
+	}
```

Comment:
Could we rewrite the function to something as below? It's more common to have a loop variable.
```go
for err := lock.TryLock(); err != nil {
     if err != lockfile.ErrBUsy {
          return err
    }
    time.Sleep(10 * time.Millisecond)
}
return nil
```